app/src/main/java/com/concordia/mcga/utilities/pathfinding/GlobalPathFinder.java

```diff
@@ -0,0 +1,251 @@
+
+    private Map<Floor, List<IndoorMapTile>> startDirections;
+    private Map<Floor, List<IndoorMapTile>> endDirections;
+
+    private Thread startIndoorDirectionsThread;
+
+    private OutdoorDirections outdoorDirections;
+
+    /**
+     * @param activity reference to the MainActivity
+     * @param startPOI
+     * @param destPOI
+     */
+    public GlobalPathFinder(MainActivity activity, POI startPOI, POI destPOI) {
+        this.startPOI = startPOI;
+        this.destPOI = destPOI;
+        this.activity = activity;
+        this.outdoorDirections = activity.getNavigationFragment().getOutdoorDirections();
+
+        this.startDirections = new HashMap<>();
+        this.endDirections = new HashMap<>();
+    }
+
+    @Override
+    public void run() {
+        //First determine type of path by analyzing POIs
+        determinePathType();
+
+        //Depending on the type of path, generate the appropriate parameters.
+        switch (pathType) {
+            case INDOOR_TO_OUTDOOR:
+                generateStartIndoorDirections();
+                break;
+            case OUTDOOR_TO_INDOOR:
+                generateDestIndoorDirections();
+                break;
+            case INDOOR_TO_OUTDOOR_TO_INDOOR:
+                generateStartIndoorDirections();
+                generateDestIndoorDirections();
+                break;
+        }
+
+        generateOutdoorDirections();
+
+        if (startIndoorDirectionsThread != null) {
+            try {
+                startIndoorDirectionsThread.join();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+
+
+        // LOGIC HERE
+        activity.notifyPathfindingComplete();
+    }
+
+    private void determinePathType() {
+        if (startPOI instanceof Room && destPOI instanceof Room) {
+            pathType = PathType.INDOOR_TO_OUTDOOR_TO_INDOOR;
+        } else if (startPOI instanceof Room && destPOI instanceof POI) {
+            pathType = PathType.INDOOR_TO_OUTDOOR;
+        } else if (startPOI instanceof POI && destPOI instanceof Room) {
+            pathType = PathType.OUTDOOR_TO_INDOOR;
+        }
+    }
+
+    private void generateOutdoorDirections() {
+        outdoorDirections.setMap(activity.getNavigationFragment().getGoogleMap());
+
+        LatLng buildingStart = null;
+        if (startPOI instanceof Room) {
+            buildingStart = ((Room) startPOI).getFloor().getBuilding().getMapCoordinates();
+        }
+        outdoorDirections.setOrigin(buildingStart);
+        outdoorDirections.setDestination(destPOI.getMapCoordinates());
+        
+        outdoorDirections.requestDirections();
+    }
+
+    private void generateStartIndoorDirections() {
+        IndoorPOI exitPortal = ((Room) startPOI).getFloor().getBuilding().getDefaultPortal();
+        GeneratePath indoorDirections = new GeneratePath((IndoorPOI) startPOI, exitPortal, true);
+        startIndoorDirectionsThread = new Thread(indoorDirections);
```

Comment:
this is the thread that is supposed to do that computation, no need to start an additional thread